### PR TITLE
cte/support_with_recursive

### PR DESCRIPTION
**Pull request by bolajiwahab**

### CODE REVIEW

**Comment by bolajiwahab** [comment]
Fixes issue https://github.com/sqlfluff/sqlfluff/issues/1087

**Comment by bolajiwahab** [comment]
@alanmcruickshank Think we can merge this now?

File: src/sqlfluff/dialects/dialect_ansi.py [hunk 1]
```diff
@@ -1803,6 +1803,7 @@ class WithCompoundStatementSegment(BaseSegment):
     match_grammar = StartsWith("WITH")
     parse_grammar = Sequence(
         "WITH",
+        Sequence("RECURSIVE", optional=True),
```
Comment (NiallRees) [inline src/sqlfluff/dialects/dialect_ansi.py:1806]:
```suggestion
        Ref.keyword("RECURSIVE", optional=True),
```

**Review by NiallRees — CHANGES_REQUESTED** [review]
This is great @bolajiwahab - just a small change needed!

**Review by NiallRees — APPROVED** [review]

**Pull request merged**